packages/@aws-cdk/aws-iam/test/merge-statements.test.ts

```diff
@@ -0,0 +1,328 @@
+import { App, Stack } from '@aws-cdk/core';
+import * as iam from '../lib';
+
+const PRINCIPAL_ARN = 'arn:aws:iam::111111111:user/user-name';
+const principal = new iam.ArnPrincipal(PRINCIPAL_ARN);
+
+const PRINCIPAL_ARN2 = 'arn:aws:iam::111111111:role/role-name';
+const principal2 = new iam.ArnPrincipal(PRINCIPAL_ARN2);
```

Comment:
Or maybe call these `userPrincipal` and `rolePrincipal`? And `PRINCIPAL_USER_ARN` and `PRINCIPAL_ROLE_ARN`?